doc/reference/ubuntu-image.rst

```diff
@@ -4,16 +4,39 @@ ubuntu-image syntax and options
 ``ubuntu-image`` is a program for generating bootable disk images. It supports building snap_-based and classical preinstalled Ubuntu images.
 
 
+Installing ``ubuntu-image``
+---------------------------
+
+The ``ubuntu-image`` tool is distributed as a snap package through the Snap Store: `ubuntu-image <https://snapcraft.io/ubuntu-image>`_. Install it either from a Snap-enabled software store provided by your distribution, or by running this command with ``sudo`` or as the ``root`` user:
+
+.. code-block::
+
+    snap install ubuntu-image --classic
+
+Note that you need have the snap daemon installed. See `Installing the daemon <https://snapcraft.io/docs/installing-snapd>`_ for instructions for your distribution.
+
+
 Snap-based images
 -----------------
 
-Snap-based images are built from a *model assertion*, which is a YAML file describing a particular combination of core, kernel, and gadget snaps, along with other declarations, signed with a digital signature asserting its authenticity.  The assets defined in the model assertion uniquely describe the device for which the image is built.
+Snap-based images are built from a *model assertion*, which is a YAML file describing a particular combination of core, kernel, and gadget snaps, along with other declarations, signed with a digital signature asserting its authenticity.  The assets defined in the model assertion uniquely describe the device for which the image is built. See `model assertion`_ for details.
 
 
 Classical images
 ----------------
 
-Classical images are built from a local `gadget tree`_ path.  The `gadget tree`_ is a primed `gadget snap`_ containing a ``gadget.yaml`` file in the ``meta`` directory and all the necessary bootloader gadget bits built. For instance, a `gadget tree`_ can be prepared by fetching a specially tailored `gadget snap`_ source and running ``snapcraft prime`` on it, with the resulting tree being stored in the ``prime/`` directory.
+Classical images are built from an image-definition YAML file, which defines the image content and resulting output format. Key elements of this configuration are the following sections:
+
+``gadget``
+    Defines how to get the gadget required to build boot assets and the partition layout.
+
+``rootfs``
+    Defines the parameters to build the content of the rootfs.
+
+``artifacts``
+    Defines the types of artefacts to create, including the actual images, manifest files, and others.
+
+Alternatively, classical images can also be built from a local `gadget tree`_ path.  The `gadget tree`_ is a primed `gadget snap`_ containing a ``gadget.yaml`` file in the ``meta`` directory and all the necessary bootloader gadget bits built. For instance, a `gadget tree`_ can be prepared by fetching a specially tailored `gadget snap`_ source and running ``snapcraft prime`` on it, with the resulting tree being stored in the ``prime/`` directory.
```

Comment:
My previous comment on this was confusing. That is not an alternative: the image definition YAML file is mandatory to build a classic image. This paragraph describes how to build the gadget assets from a "gadget tree". This is only a part of what is built by ubuntu-image.

Maybe this paragraph would make more sense in the `gadget` section in the documentation of the image definition. As you can see in https://github.com/canonical/ubuntu-image/blob/main/internal/imagedefinition/README.rst, the user can choose a `type`. This paragraph is a semi-detailed explanation of what could be done to get a valid gadget tree if the user selected `type: prebuilt`. But since ubuntu-image now handles this process pretty well, most of our users should probably use `type: git` and let ubuntu-image do the work.

Reply:
> My previous comment on this was confusing. That is not an alternative: the image definition YAML file is mandatory to build a classic image.

Thank you. I need to find the time to actually use the tool, so that I don't just blindly blunder about... :/

I'll add the paragraph to the image-definition doc.